oggm/core/preprocessing/gis.py

```diff
@@ -383,60 +390,61 @@ def define_glacier_region(gdir, entity=None):
 
     # Open DEM
     source = entity.DEM_SOURCE if hasattr(entity, 'DEM_SOURCE') else None
-    dem_file, dem_source = get_topo_file((minlon, maxlon), (minlat, maxlat),
+    dem_list, dem_source = get_topo_file((minlon, maxlon), (minlat, maxlat),
                                          rgi_region=gdir.rgi_region,
                                          source=source)
     log.debug('%s: DEM source: %s', gdir.rgi_id, dem_source)
-    dem = gdal.Open(dem_file)
-    geo_t = dem.GetGeoTransform()
 
-    # Input proj
-    dem_proj = dem.GetProjection()
-    if dem_proj == '':
-        # Assume defaults
-        wgs84 = osr.SpatialReference()
-        wgs84.ImportFromEPSG(4326)
-        dem_proj = wgs84.ExportToWkt()
-
-
-    # Dest proj
-    gproj = osr.SpatialReference()
-    gproj.SetProjCS('Local transverse Mercator')
-    gproj.SetWellKnownGeogCS("WGS84")
-    gproj.SetTM(np.float(0), gdir.cenlon,
-                np.float(0.9996), np.float(0), np.float(0))
-
-    # Create an in-memory raster
-    mem_drv = gdal.GetDriverByName('MEM')
-
-    # GDALDataset Create (char *pszName, int nXSize, int nYSize,
-    #                     int nBands, GDALDataType eType)
-    dest = mem_drv.Create('', nx, ny, 1, gdal.GDT_Float32)
-
-    # Calculate the new geotransform
-    new_geo = (ulx, dx, geo_t[2], uly, geo_t[4], -dx)
-
-    # Set the geotransform
-    dest.SetGeoTransform(new_geo)
-    dest.SetProjection(gproj.ExportToWkt())
-
-    # Perform the projection/resampling
+    # A glacier area can cover than one tile:
+    if len(dem_list) == 1:
+        dem = rasterio.open(dem_list[0])
+    else:
+        demraster = [rasterio.open(s) for s in dem_list]
+        dem, output_transform = merge_tool(demraster)
+        profile = demraster[0].profile
+        if 'affine' in profile:
+            profile.pop('affine')
+        profile['transform'] = output_transform
+        profile['height'] = dem.shape[0]
+        profile['width'] = dem.shape[1]
+        profile['driver'] = 'GTiff'
```

Comment:
Are all these ``profile`` stuff necessary? A couple of lines later you overwrite it. Or does changing the profile dict play some role here?